ethcore/engines/authority-round/src/lib.rs

```diff
@@ -1377,6 +1380,25 @@ impl Engine for AuthorityRound {
 			Err(EngineError::DoubleVote(*header.author()))?;
 		}
 
+		// Report malice if the validator produced other sibling blocks in the same step.
+		let received_step_key = (step, *header.author());
+		let new_hash = header.hash();
+		if self.received_step_hashes.read().get(&received_step_key).into_iter().any(|h| *h != new_hash) {
```

Comment:
I must be missing something, `self.received_step_hashes.read().get(&received_step_key)` returns a single `H256` yes? Why not `if self.received_step_hashes.read().get(&received_step_key) != Some(&new_hash)`?

Reply:
`a != Some(b)` is `true` if `a` is `None`.
`a.into_iter().any(|h| h != b)` is `false` if `a` is `None`.

We could use `a.map_or(false, |h| h != b)`. Not sure if that's more readable. After all, we want to say something like: "If the map already contains a hash, _and_ that's different from `new_hash`, report the block author."

Reply:
IMHO, `map_or` is more readable in this context because when using one element it may be confusing to the reader